sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/MetadataOnlyTable.java

```diff
@@ -0,0 +1,231 @@
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * A concrete {@code Table} implementation that only contains the table metadata without
+ * implementing read/write directly. It represents a general Spark data source table or
+ * a Spark view, and relies on Spark to interpret the table metadata, resolve the table
+ * provider into a data source, or read it as a view.
+ *
+ * @since 4.1.0
+ */
+@Evolving
+public class MetadataOnlyTable implements Table {
+  private final Builder builder;
+
+  private MetadataOnlyTable(Builder builder) {
+    this.builder = builder;
+  }
+
+  public static class Builder {
+    private final Column[] columns;
+    private String name = "data_source_table_or_view";
+    private String provider = null;
+    private String location = null;
+    private String tableType = null;
+    private String owner = null;
+    private String comment = null;
+    private String collation = null;
+    private String viewText = null;
+    private String createVersion = "";
+    private long createTime = 0;
+    private Map<String, String> tableProps = Map.ofEntries();
+    private Map<String, String> serdeProps = Map.ofEntries();
+    private Transform[] partitioning = new Transform[0];
+
+    public Builder(Column[] columns) {
+      assert columns != null;
+      this.columns = columns;
+    }
+
+    public Builder(StructType schema) {
+      assert schema != null;
+      this.columns = CatalogV2Util.structTypeToV2Columns(schema);
+    }
+
+    public Builder withName(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder withProvider(String provider) {
+      this.provider = provider;
+      return this;
+    }
+
+    public Builder withLocation(String location) {
+      this.location = location;
+      return this;
+    }
+
+    public Builder withTableType(String tableType) {
+      this.tableType = tableType;
+      return this;
+    }
+
+    public Builder withOwner(String owner) {
+      this.owner = owner;
+      return this;
+    }
+
+    public Builder withComment(String comment) {
+      this.comment = comment;
+      return this;
+    }
+
+    public Builder withCollation(String collation) {
+      this.collation = collation;
+      return this;
+    }
+
+    public Builder withViewText(String viewText) {
+      this.viewText = viewText;
+      this.tableType = TableSummary.VIEW_TABLE_TYPE;
+      return this;
+    }
+
+    public Builder withCreateVersion(String createVersion) {
+      this.createVersion = createVersion;
+      return this;
+    }
+
+    public Builder withCreateTime(long createTime) {
+      this.createTime = createTime;
+      return this;
+    }
+
+    public Builder withTableProps(Map<String, String> tableProps) {
+      this.tableProps = tableProps;
+      return this;
+    }
+
+    public Builder withSerdeProps(Map<String, String> serdeProps) {
```

Comment:
Just an opinion, serde is quite bound to HMS (?) and the newer DSV2 dont have that, will it make sense to abstract this (maybe something like 'storageProperties')?  This is just for HMS table support?